### PR TITLE
Adding remote sensing packages

### PR DESCRIPTION
**Pull request by petebunting**
Fixed cmake by updating checksum
Updated version of py-statsmodels

Adding builds for remote sensing data processing software such as RSGISLib (https://www.rsgislib.org), ARCSI (https://arcsi.remotesensing.info) and EODataDown (for automating the downloading and processing of EO data):

py-arcsi
py-compute-job-recorder
py-eodatadown
py-opendatacube
py-planet
py-py6s
py-python-fmask
py-rios
rsgislib
sixs

This required the following additional dependencies:

py-cachetools
py-deap
py-gitdb
py-gitpython
py-google-auth
py-google-cloud
py-google-resumable-media-python
py-ml-tooling
py-pyod
py-pypeg2
py-pysolar
py-smmap
py-stopit
py-tpot

### CODE REVIEW

File: var/spack/repos/builtin/packages/cmake/package.py [hunk 1]
```diff
@@ -13,7 +13,7 @@ class Cmake(Package):
     url      = 'https://github.com/Kitware/CMake/releases/download/v3.13.0/cmake-3.13.0.tar.gz'
     maintainers = ['chuckatkins']
 
-    version('3.15.1', sha256='cdee72c5ef934c4972f1ad4e9c35712589345f3470a3cf15f7be493c170cc965')
+    version('3.15.1', sha256='18dec548d8f8b04d53c60f9cedcebaa6762f8425339d1e2c889c383d3ccdd7f7')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/cmake/package.py]: @chuckatkins can you confirm that this checksum is correct?

File: var/spack/repos/builtin/packages/py-arcsi/package.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-arcsi/package.py]: This is the old Spack license. You'll need to update to the new license for your contribution to be accepted.

File: var/spack/repos/builtin/packages/py-arcsi/package.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+#
+#     spack install py-arcsi
+#
+# You can edit this file again by typing:
+#
+#     spack edit py-arcsi
+#
+# See the Spack documentation for more information on packaging.
+#
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-arcsi/package.py]: These FIXME comments can be removed

File: var/spack/repos/builtin/packages/py-arcsi/package.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+       topographic correction etc. of Earth Observation optical imagery (Blue-SWIR). 
+       The aim of ARCSI is to provide as automatic as possible method of generating 
+       Analysis Ready Data (ARD)."""
+
+    homepage = "https://www.arcsi.remotesensing.info"
+    url      = "https://bitbucket.org/petebunting/arcsi/downloads/arcsi-3.1.6.tar.gz"
+
+    maintainers = ['petebunting']
+
+    version('3.2.3a', '177aacacdae897b43c4a717c88e95ea7')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-arcsi/package.py]: Is this an alpha release? If so, we may want to add `preferred=True` to the latest stable release.

File: var/spack/repos/builtin/packages/py-arcsi/package.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    depends_on('python', type=('build', 'run'))
+    depends_on('rsgislib', type=('build', 'run'))
+    depends_on('py-py6s', type=('build', 'run'))
+    depends_on('py-rios', type=('build', 'run'))
+    depends_on('py-scikit-learn', type=('build', 'run'))
+    
+    def install(self, spec, prefix):
+        import subprocess
+        cmd = '{0} setup.py install --prefix={1}'.format(spec['python'].command.path, prefix)
+        subprocess.call(cmd, shell=True)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-arcsi/package.py]: The default `build` and `install` methods for the `PythonPackage` base class should handle everything. Did you encounter a problem with them?

**Comment by adamjstewart** [comment]
It also makes things easier to review if you split each package into a separate PR: https://spack.readthedocs.io/en/latest/contribution_guide.html

Not a requirement, just a suggestion to speed things along. For example, all of your packages could be good, but if the `cmake` checksum change is controversial, nothing else will get merged.

File: var/spack/repos/builtin/packages/cmake/package.py (hunk 1 above)
Comment (chuckatkins) [reply var/spack/repos/builtin/packages/cmake/package.py]: This has already been fixed and merged in #12182

**Comment by chuckatkins** [comment]
@adamjstewart since these are all new packages, should the older versions be dropped if they're not being used at all?

**Comment by adamjstewart** [comment]
@chuckatkins that's up to @petebunting. I generally don't bother adding older versions of packages unless I need them. If @petebunting doesn't need them, they can be removed.

File: var/spack/repos/builtin/packages/cmake/package.py (hunk 1 above)
Comment (petebunting) [reply var/spack/repos/builtin/packages/cmake/package.py]: Thanks for that, I've updated and removed this from the pull request.

File: var/spack/repos/builtin/packages/py-arcsi/package.py (hunk 2 above)
Comment (petebunting) [reply var/spack/repos/builtin/packages/py-arcsi/package.py]: I've updated them all with the new license - I'd copied these in from my own repo of packages I'd created last year.

File: var/spack/repos/builtin/packages/py-arcsi/package.py (hunk 3 above)
Comment (petebunting) [reply var/spack/repos/builtin/packages/py-arcsi/package.py]: Done.

File: var/spack/repos/builtin/packages/py-arcsi/package.py (hunk 5 above)
Comment (petebunting) [reply var/spack/repos/builtin/packages/py-arcsi/package.py]: I've sorted this out, I'd imported setuptools but didn't need it so was getting an error about --single-version-externally-managed so this is how I'd solved it last year but had put an incorrect dependency into the files.

**Comment by petebunting** [comment]
I've removed the older versions.

**Comment by adamjstewart** [comment]
@petebunting I added several more Google Earth Engine packages that add some of the same dependencies you did. Make sure you're using the PyPI names, not the GitHub names. Can you rebase and resolve the flake8 warnings:
```
var/spack/repos/builtin/packages/py-arcsi/package.py:21: [W291] trailing whitespace
var/spack/repos/builtin/packages/py-arcsi/package.py:22: [E501] line too long (81 > 79 characters)
var/spack/repos/builtin/packages/py-arcsi/package.py:22: [W291] trailing whitespace
var/spack/repos/builtin/packages/py-arcsi/package.py:23: [W291] trailing whitespace
var/spack/repos/builtin/packages/py-arcsi/package.py:24: [E501] line too long (84 > 79 characters)
var/spack/repos/builtin/packages/py-arcsi/package.py:24: [W291] trailing whitespace
var/spack/repos/builtin/packages/py-arcsi/package.py:25: [E501] line too long (83 > 79 characters)
var/spack/repos/builtin/packages/py-arcsi/package.py:25: [W291] trailing whitespace
var/spack/repos/builtin/packages/py-arcsi/package.py:41: [W293] blank line contains whitespace
var/spack/repos/builtin/packages/py-arcsi/package.py:41: [W391] blank line at end of file
var/spack/repos/builtin/packages/py-cachetools/package.py:22: [W291] trailing whitespace
var/spack/repos/builtin/packages/py-cachetools/package.py:23: [E501] line too long (86 > 79 characters)
var/spack/repos/builtin/packages/py-cachetools/package.py:23: [W291] trailing whitespace
var/spack/repos/builtin/packages/py-cachetools/package.py:37: [W391] blank line at end of file
var/spack/repos/builtin/packages/py-compute-job-recorder/package.py:21: [E501] line too long (94 > 79 characters)
var/spack/repos/builtin/packages/py-compute-job-recorder/package.py:32: [W293] blank line contains whitespace
var/spack/repos/builtin/packages/py-compute-job-recorder/package.py:33: [W391] blank line at end of file
var/spack/repos/builtin/packages/py-deap/package.py:29: [W391] blank line at end of file
var/spack/repos/builtin/packages/py-eodatadown/package.py:19: [E302] expected 2 blank lines, found 1
var/spack/repos/builtin/packages/py-eodatadown/package.py:20: [E501] line too long (86 > 79 characters)
var/spack/repos/builtin/packages/py-eodatadown/package.py:41: [W293] blank line contains whitespace
var/spack/repos/builtin/packages/py-eodatadown/package.py:42: [W391] blank line at end of file
var/spack/repos/builtin/packages/py-gitdb/package.py:21: [E501] line too long (101 > 79 characters)
var/spack/repos/builtin/packages/py-gitdb/package.py:30: [W391] blank line at end of file
var/spack/repos/builtin/packages/py-gitpython/package.py:19: [E302] expected 2 blank lines, found 1
var/spack/repos/builtin/packages/py-gitpython/package.py:29: [W391] blank line at end of file
var/spack/repos/builtin/packages/py-google-auth/package.py:40: [W391] blank line at end of file
var/spack/repos/builtin/packages/py-google-cloud/package.py:21: [E501] line too long (117 > 79 characters)
var/spack/repos/builtin/packages/py-google-cloud/package.py:39: [E303] too many blank lines (2)
var/spack/repos/builtin/packages/py-google-cloud/package.py:43: [W293] blank line contains whitespace
var/spack/repos/builtin/packages/py-google-cloud/package.py:46: [E501] line too long (138 > 79 characters)
var/spack/repos/builtin/packages/py-google-cloud/package.py:49: [W293] blank line contains whitespace
var/spack/repos/builtin/packages/py-google-cloud/package.py:52: [E501] line too long (138 > 79 characters)
var/spack/repos/builtin/packages/py-google-cloud/package.py:55: [W293] blank line contains whitespace
var/spack/repos/builtin/packages/py-google-cloud/package.py:58: [E501] line too long (138 > 79 characters)
var/spack/repos/builtin/packages/py-google-cloud/package.py:61: [W293] blank line contains whitespace
var/spack/repos/builtin/packages/py-google-cloud/package.py:64: [E501] line too long (138 > 79 characters)
var/spack/repos/builtin/packages/py-google-cloud/package.py:67: [W293] blank line contains whitespace
var/spack/repos/builtin/packages/py-google-cloud/package.py:70: [E501] line too long (138 > 79 characters)
var/spack/repos/builtin/packages/py-google-cloud/package.py:73: [W293] blank line contains whitespace
var/spack/repos/builtin/packages/py-google-cloud/package.py:76: [E501] line too long (138 > 79 characters)
var/spack/repos/builtin/packages/py-google-cloud/package.py:79: [W293] blank line contains whitespace
var/spack/repos/builtin/packages/py-google-cloud/package.py:82: [E501] line too long (138 > 79 characters)
var/spack/repos/builtin/packages/py-google-cloud/package.py:86: [W391] blank line at end of file
var/spack/repos/builtin/packages/py-google-resumable-media-python/package.py:34: [W293] blank line contains whitespace
var/spack/repos/builtin/packages/py-google-resumable-media-python/package.py:36: [W391] blank line at end of file
var/spack/repos/builtin/packages/py-ml-tooling/package.py:35: [W391] blank line at end of file
var/spack/repos/builtin/packages/py-opendatacube/package.py:21: [E501] line too long (88 > 79 characters)
var/spack/repos/builtin/packages/py-opendatacube/package.py:52: [W391] blank line at end of file
var/spack/repos/builtin/packages/py-planet/package.py:26: [W293] blank line contains whitespace
var/spack/repos/builtin/packages/py-planet/package.py:33: [W391] blank line at end of file
var/spack/repos/builtin/packages/py-py6s/package.py:22: [W291] trailing whitespace
var/spack/repos/builtin/packages/py-py6s/package.py:23: [W291] trailing whitespace
var/spack/repos/builtin/packages/py-py6s/package.py:41: [W391] blank line at end of file
var/spack/repos/builtin/packages/py-pyod/package.py:36: [W391] blank line at end of file
var/spack/repos/builtin/packages/py-pypeg2/package.py:27: [W391] blank line at end of file
var/spack/repos/builtin/packages/py-pysolar/package.py:19: [E302] expected 2 blank lines, found 1
var/spack/repos/builtin/packages/py-pysolar/package.py:20: [W291] trailing whitespace
var/spack/repos/builtin/packages/py-pysolar/package.py:21: [W291] trailing whitespace
var/spack/repos/builtin/packages/py-pysolar/package.py:30: [W391] blank line at end of file
var/spack/repos/builtin/packages/py-python-fmask/package.py:19: [E302] expected 2 blank lines, found 1
var/spack/repos/builtin/packages/py-python-fmask/package.py:20: [W291] trailing whitespace
var/spack/repos/builtin/packages/py-python-fmask/package.py:33: [W293] blank line contains whitespace
var/spack/repos/builtin/packages/py-python-fmask/package.py:35: [W391] blank line at end of file
var/spack/repos/builtin/packages/py-rios/package.py:19: [E302] expected 2 blank lines, found 1
var/spack/repos/builtin/packages/py-rios/package.py:20: [W291] trailing whitespace
var/spack/repos/builtin/packages/py-rios/package.py:21: [W291] trailing whitespace
var/spack/repos/builtin/packages/py-rios/package.py:22: [E501] line too long (81 > 79 characters)
var/spack/repos/builtin/packages/py-rios/package.py:22: [W291] trailing whitespace
var/spack/repos/builtin/packages/py-rios/package.py:23: [E501] line too long (83 > 79 characters)
var/spack/repos/builtin/packages/py-rios/package.py:23: [W291] trailing whitespace
var/spack/repos/builtin/packages/py-rios/package.py:35: [W293] blank line contains whitespace
var/spack/repos/builtin/packages/py-rios/package.py:36: [W391] blank line at end of file
var/spack/repos/builtin/packages/py-smmap/package.py:21: [E501] line too long (184 > 79 characters)
var/spack/repos/builtin/packages/py-smmap/package.py:29: [W391] blank line at end of file
var/spack/repos/builtin/packages/py-stopit/package.py:21: [E501] line too long (146 > 79 characters)
var/spack/repos/builtin/packages/py-stopit/package.py:29: [W391] blank line at end of file
var/spack/repos/builtin/packages/py-tpot/package.py:21: [E501] line too long (119 > 79 characters)
var/spack/repos/builtin/packages/py-tpot/package.py:37: [W391] blank line at end of file
var/spack/repos/builtin/packages/rsgislib/package.py:21: [E302] expected 2 blank lines, found 1
var/spack/repos/builtin/packages/rsgislib/package.py:22: [W291] trailing whitespace
var/spack/repos/builtin/packages/rsgislib/package.py:23: [E501] line too long (83 > 79 characters)
var/spack/repos/builtin/packages/rsgislib/package.py:29: [W293] blank line contains whitespace
var/spack/repos/builtin/packages/rsgislib/package.py:48: [W293] blank line contains whitespace
var/spack/repos/builtin/packages/rsgislib/package.py:50: [W293] blank line contains whitespace
var/spack/repos/builtin/packages/rsgislib/package.py:52: [N806] variable 'cmdPython' in function should be lowercase
var/spack/repos/builtin/packages/rsgislib/package.py:54: [N806] variable 'cmdBoostLib' in function should be lowercase
var/spack/repos/builtin/packages/rsgislib/package.py:55: [N806] variable 'cmdBoostInc' in function should be lowercase
var/spack/repos/builtin/packages/rsgislib/package.py:55: [E501] line too long (86 > 79 characters)
var/spack/repos/builtin/packages/rsgislib/package.py:57: [N806] variable 'cmdGDALLib' in function should be lowercase
var/spack/repos/builtin/packages/rsgislib/package.py:58: [N806] variable 'cmdGDALInc' in function should be lowercase
var/spack/repos/builtin/packages/rsgislib/package.py:58: [E501] line too long (82 > 79 characters)
var/spack/repos/builtin/packages/rsgislib/package.py:60: [N806] variable 'cmdHDF5Lib' in function should be lowercase
var/spack/repos/builtin/packages/rsgislib/package.py:61: [N806] variable 'cmdHDF5Inc' in function should be lowercase
var/spack/repos/builtin/packages/rsgislib/package.py:61: [E501] line too long (82 > 79 characters)
var/spack/repos/builtin/packages/rsgislib/package.py:63: [N806] variable 'cmdXercescLib' in function should be lowercase
var/spack/repos/builtin/packages/rsgislib/package.py:63: [E501] line too long (85 > 79 characters)
var/spack/repos/builtin/packages/rsgislib/package.py:64: [N806] variable 'cmdXercescInc' in function should be lowercase
var/spack/repos/builtin/packages/rsgislib/package.py:64: [E501] line too long (92 > 79 characters)
var/spack/repos/builtin/packages/rsgislib/package.py:66: [N806] variable 'cmdGSLLib' in function should be lowercase
var/spack/repos/builtin/packages/rsgislib/package.py:67: [N806] variable 'cmdGSLInc' in function should be lowercase
var/spack/repos/builtin/packages/rsgislib/package.py:69: [N806] variable 'cmdGEOSLib' in function should be lowercase
var/spack/repos/builtin/packages/rsgislib/package.py:70: [N806] variable 'cmdGEOSInc' in function should be lowercase
var/spack/repos/builtin/packages/rsgislib/package.py:70: [E501] line too long (82 > 79 characters)
var/spack/repos/builtin/packages/rsgislib/package.py:72: [N806] variable 'cmdMuParserLib' in function should be lowercase
var/spack/repos/builtin/packages/rsgislib/package.py:72: [E501] line too long (87 > 79 characters)
var/spack/repos/builtin/packages/rsgislib/package.py:73: [N806] variable 'cmdMuParserInc' in function should be lowercase
var/spack/repos/builtin/packages/rsgislib/package.py:73: [E501] line too long (94 > 79 characters)
var/spack/repos/builtin/packages/rsgislib/package.py:75: [N806] variable 'cmdCGALLib' in function should be lowercase
var/spack/repos/builtin/packages/rsgislib/package.py:76: [N806] variable 'cmdCGALInc' in function should be lowercase
var/spack/repos/builtin/packages/rsgislib/package.py:76: [E501] line too long (82 > 79 characters)
var/spack/repos/builtin/packages/rsgislib/package.py:78: [N806] variable 'cmdGMPLib' in function should be lowercase
var/spack/repos/builtin/packages/rsgislib/package.py:79: [N806] variable 'cmdGMPInc' in function should be lowercase
var/spack/repos/builtin/packages/rsgislib/package.py:81: [N806] variable 'cmdMPFRLib' in function should be lowercase
var/spack/repos/builtin/packages/rsgislib/package.py:82: [N806] variable 'cmdMPFRInc' in function should be lowercase
var/spack/repos/builtin/packages/rsgislib/package.py:82: [E501] line too long (82 > 79 characters)
var/spack/repos/builtin/packages/rsgislib/package.py:84: [F841] local variable 'cmdKEALib' is assigned to but never used
var/spack/repos/builtin/packages/rsgislib/package.py:84: [N806] variable 'cmdKEALib' in function should be lowercase
var/spack/repos/builtin/packages/rsgislib/package.py:85: [N806] variable 'cmdKEAInc' in function should be lowercase
var/spack/repos/builtin/packages/rsgislib/package.py:85: [E501] line too long (82 > 79 characters)
var/spack/repos/builtin/packages/rsgislib/package.py:86: [W293] blank line contains whitespace
var/spack/repos/builtin/packages/rsgislib/package.py:87: [E226] missing whitespace around arithmetic operator
var/spack/repos/builtin/packages/rsgislib/package.py:87: [E501] line too long (441 > 79 characters)
var/spack/repos/builtin/packages/rsgislib/package.py:88: [W293] blank line contains whitespace
var/spack/repos/builtin/packages/rsgislib/package.py:92: [W293] blank line contains whitespace
var/spack/repos/builtin/packages/rsgislib/package.py:94: [W291] trailing whitespace
var/spack/repos/builtin/packages/rsgislib/package.py:95: [W293] blank line contains whitespace
var/spack/repos/builtin/packages/rsgislib/package.py:95: [W391] blank line at end of file
var/spack/repos/builtin/packages/sixs/package.py:31: [W293] blank line contains whitespace
var/spack/repos/builtin/packages/sixs/package.py:36: [W293] blank line contains whitespace
var/spack/repos/builtin/packages/sixs/package.py:37: [W291] trailing whitespace
var/spack/repos/builtin/packages/sixs/package.py:39: [E226] missing whitespace around arithmetic operator
var/spack/repos/builtin/packages/sixs/package.py:41: [W293] blank line contains whitespace
var/spack/repos/builtin/packages/sixs/package.py:45: [W391] blank line at end of file
```

**Comment by alalazo** [comment]
Closing as stale. Feel free to reopen if you want to push this PR forward.

**Comment by Sinan81** [comment]
I might be able to help with this PR. Will create separate PRs for each package as suggested by adam.

**Comment by Sinan81** [comment]
@petebunting the bitbucket links to source files seems to be broken due to bitbucket discontinuing mercurial repo stuff. Just to let you know.

**Pull request closed without merging**